codec-redis/src/main/java/io/netty/handler/codec/redis/RedisEncoder.java

```diff
@@ -0,0 +1,201 @@
+            ByteBuf buf = allocator.ioBuffer(RedisConstants.TYPE_LENGTH + RedisConstants.NULL_LENGTH +
+                                             RedisConstants.EOL_LENGTH);
+            buf.writeByte(RedisMessageType.BULK_STRING.value());
+            buf.writeShort(RedisConstants.NULL_SHORT);
+            buf.writeShort(RedisConstants.EOL_SHORT);
+            out.add(buf);
+        } else {
+            ByteBuf headerBuf = allocator.ioBuffer(RedisConstants.TYPE_LENGTH + RedisConstants.LONG_MAX_LENGTH +
+                                                   RedisConstants.EOL_LENGTH);
+            headerBuf.writeByte(RedisMessageType.BULK_STRING.value());
+            headerBuf.writeBytes(numberToBytes(msg.content().readableBytes()));
+            headerBuf.writeShort(RedisConstants.EOL_SHORT);
+            out.add(headerBuf);
+            out.add(msg.content().retain());
+            out.add(Unpooled.copyShort(RedisConstants.EOL_SHORT));
+        }
+    }
+
+    /**
+     * Write array header only without body. Use this if you want to write arrays as streaming.
+     */
+    private void writeArrayHeader(ByteBufAllocator allocator, ArrayHeaderRedisMessage msg, List<Object> out) {
+        writeArrayHeader(allocator, msg.isNull(), msg.length(), out);
+    }
+
+    /**
+     * Write full constructed array message.
+     */
+    private void writeArrayMessage(ByteBufAllocator allocator, ArrayRedisMessage msg, List<Object> out) {
+        if (msg.isNull()) {
+            writeArrayHeader(allocator, msg.isNull(), RedisConstants.NULL_VALUE, out);
+        } else {
+            writeArrayHeader(allocator, msg.isNull(), msg.children().size(), out);
+            for (RedisMessage child : msg.children()) {
+                writeRedisMessage(allocator, child, out);
+            }
+        }
+    }
+
+    private void writeArrayHeader(ByteBufAllocator allocator, boolean isNull, long length, List<Object> out) {
+        if (isNull) {
+            final ByteBuf buf = allocator.ioBuffer(RedisConstants.TYPE_LENGTH + RedisConstants.NULL_LENGTH +
+                                                   RedisConstants.EOL_LENGTH);
+            buf.writeByte(RedisMessageType.ARRAY_HEADER.value());
+            buf.writeShort(RedisConstants.NULL_SHORT);
+            buf.writeShort(RedisConstants.EOL_SHORT);
+            out.add(buf);
+        } else {
+            final ByteBuf buf = allocator.ioBuffer(RedisConstants.TYPE_LENGTH + RedisConstants.LONG_MAX_LENGTH +
+                                                   RedisConstants.EOL_LENGTH);
+            buf.writeByte(RedisMessageType.ARRAY_HEADER.value());
+            buf.writeBytes(numberToBytes(length));
+            buf.writeShort(RedisConstants.EOL_SHORT);
+            out.add(buf);
+        }
+    }
+
+    private byte[] numberToBytes(long value) {
+        byte[] bytes = messagePool.getByteBufOfInteger(value);
+        return bytes != null ? bytes : Long.toString(value).getBytes(CharsetUtil.US_ASCII);
```

Comment:
I think we will create an optimized version of this conversion in the future (as previously mentioned).